lib/actions/preload.coffee

```diff
@@ -203,16 +202,23 @@ module.exports =
 					.then (commit) ->
 
 						# No commit specified => use the latest commit
-						if commit != LATEST
+						if commit == LATEST
+							options.commit = application.commit
+						else
 							options.commit = commit
 
 						# Propose to disable automatic app updates if the commit is not the latest
 						offerToDisableAutomaticUpdates(Promise, form, resin, application, commit)
-			.then ->
-
-				# All options are ready: preload the image.
-				preload.run(resin, docker, options)
-				.catch(preload.errors.ResinError, expectedError)
+				.then ->
+
+					builds = builds.map (build) ->
+						build.slice(-preload.BUILD_HASH_LENGTH)
+					if options.commit in builds
+						console.log('This build is already preloaded in this image.')
+						process.exit(0)
+					# All options are ready: preload the image.
+					preload.run(resin, docker, options)
+					.catch(preload.errors.ResinError, expectedError)
```

Comment:
Doesn't matter immediately for this PR, but I'd really like a `ResinError` subclass/flag for errors that are issues in usage or environment, not functional unknown errors, so we can filter for those and behave differently (this would basically replace `expectedError`, and instead just filter errors in our standard error handling to handle that case differently).

We can ignore this for now - just a note for the future.